webui/react/src/routes/utils.ts

```diff
@@ -0,0 +1,118 @@
+import { pathToRegexp } from 'path-to-regexp';
+import { MouseEvent, MouseEventHandler } from 'react';
+
+import history from 'routes/history';
+import { Command, CommandType } from 'types';
+import { clone } from 'utils/data';
+
+import routes from './routes';
+
+export const serverAddress = (avoidDevProxy = false, path = ''): string => {
+  const address = avoidDevProxy && process.env.IS_DEV
+    ? 'http://localhost:8080'
+    : `${window.location.protocol}//${window.location.host}`;
+  return address + path;
```

Comment:
nit: since we now have the path option what are your thoughts on using helpers we have below to ensure the passed in `path` is an absolute path and not a full path.